js/src/hooks/useStoreAddress.js

```diff
@@ -66,7 +66,11 @@ export default function useStoreAddress( source = 'wc' ) {
 
 		const [ address, address2 = '' ] = streetAddress.split( '\n' );
 		const country = countryNameDict[ storeAddress?.country ];
-		const isAddressFilled = !! ( address && city && country && postcode );
+		const isAddressFilled =
+			Array.isArray( contact.has_address_errors ) &&
+			contact.has_address_errors.length > 0
+				? false
+				: true;
```

Comment:
💅  This was a bit confusing for me at first glance. 

❓  Why we need `Array.isArray( contact.has_address_errors )` as far as I read the code in Backend we are using types and we are returning always an array. So this made me thing... mmm It could be the case other type than array is returned? 

💅  

`const isAddressFilled = contact.has_address_errors.length > 0 ? false : true;`

I'd change this to... 

`const isAddressFilled = ! contact.wc_address_errors.length;`

See also my comment [here](https://github.com/woocommerce/google-listings-and-ads/pull/1169/files#r778182747) regarding  the `has_address_errors` name.

Reply:
I think I went too cautious here doing that double checking with the array. See commit: [7b1c89b](https://github.com/woocommerce/google-listings-and-ads/pull/1169/commits/7b1c89bf566dfd9c097f84ac6e9d256a54f7b584)